src/lib/numbers.ts

```diff
@@ -10,10 +10,21 @@ export const BIG_NUMBERS = {
   ONE: new BigNumber(1),
 };
 
+// defaults to zero if null or empty
 export const MustBigNumber = (amount?: BigNumberish | null): BigNumber =>
   // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
   new BigNumber(amount || 0);
 
+// undefined if falsey otherwise a valid bignumber
+export const MaybeBigNumber = (amount?: BigNumberish | null): BigNumber | undefined =>
+  amount ? MustBigNumber(amount) : undefined;
+
+// doesnt allow null, always returns big number
+// empty string becomes null though
+export const ToBigNumber = (amount: BigNumberish): BigNumber => {
+  return MustBigNumber(amount);
+};
```

Comment:
Wouldn't this throw or produce some sort of big number NaN? What is the advantage of using this versus `mustBigNumber`?

Reply:
MustBigNumber allows null/undefined, this doesn't

Reply:
why will this produce NaN?

Reply:
not sure if it would produce NaN but BigNumberish could be a string which may not be a number. But what I'm wondering is if we **know** `amount` will always be defined why not use `MustBigNumber`? Or are we hoping to catch if this throws?

Reply:
it's more for type safety. By calling toBignumber you are saying "hey i'm pretty sure this is always defined, compile error if it's actually nullable"